utils/filesystem/files_test.go

```diff
@@ -133,6 +130,64 @@ func TestOpen(t *testing.T) {
 	}
 }
 
+func TestTouch(t *testing.T) {
+	// Similar to https://github.com/m-spitfire/go-touch/blob/master/touch_test.go
+	for _, fsType := range FileSystemTypes {
+		t.Run(fmt.Sprintf("%v_for_fs_%v", t.Name(), fsType), func(t *testing.T) {
+			fs := NewFs(fsType)
+			tmpDir, err := fs.TempDirInTempDir("test-touch-")
+			require.NoError(t, err)
+			defer func() { _ = fs.Rm(tmpDir) }()
+			tmpFile, err := fs.TouchTempFile(tmpDir, "test-touch*.test")
+			require.NoError(t, err)
+			assert.True(t, fs.Exists(tmpFile))
+			empty, _ := fs.IsEmpty(tmpFile)
+			assert.True(t, empty)
+			timesFile1, err := fs.StatTimes(tmpFile)
+			require.NoError(t, err)
+
+			tmpFile2 := filepath.Join(tmpDir, "testtouch2.test")
+			err = fs.Touch(tmpFile2)
+			require.NoError(t, err)
+			empty, _ = fs.IsEmpty(tmpFile2)
```

Comment:
I'd probably keep the error and require it as noerror. For example, If touch had caused a permission change you would get an error and empty would be false but the false would be due to an error not because its empty etc.

Reply:
fair enough